test/Gen.hs

```diff
@@ -105,9 +117,15 @@ s = genT
 type S = T "S"
 
 w :: Gen W
-w = list (linear 0 100) genT
+w = genT
+
+type W = T "W"
+
+fn :: (Fn.Arg a, Fn.Vary a, Show a) => Gen b -> Gen (a -> b)
+fn b = Gen (lift (fmap (fmap runTerm) . showingFn <$> Fn.fn (fst <$> runWriterT (runGen b))))
```

Comment:
This definition regrettably discards labels, because there’s no way to plumb them through `Fn.fn` or its limitation to `Hedgehog.Gen`.